animal-sniffer/src/main/java/org/codehaus/mojo/animal_sniffer/ClassFileVisitor.java

```diff
@@ -155,15 +151,23 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) {
                 return FileVisitResult.CONTINUE;
             }
 
+            @Override
+            public FileVisitResult postVisitDirectory(final Path dir, final IOException exc) throws IOException {
+                for (final Path file : files) {
+                    try (final InputStream inputStream = Files.newInputStream(file)) {
+                        process(file.toString(), inputStream);
+                    }
+                }
+                files.clear();
+                return super.postVisitDirectory(dir, exc);
```

Comment:
Might be better to simply return `CONTINUE` here once exception checking is done at the beginning of the method, since the [parent implementation](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/file/SimpleFileVisitor.html#postVisitDirectory(T,java.io.IOException)) behaves the same way.